packages/components/src/select-control/index.tsx

```diff
@@ -66,12 +66,14 @@ function UnforwardedSelectControl(
 			const selectedOptions = Array.from( event.target.options ).filter(
 				( { selected } ) => selected
 			);
-			const newValues = selectedOptions.map( ( { value } ) => value );
+			const newValues = selectedOptions.map(
+				( { value } ) => value as T
+			);
 			props.onChange?.( newValues, { event } );
 			return;
 		}
 
-		props.onChange?.( event.target.value, { event } );
+		props.onChange?.( event.target.value as T, { event } );
```

Comment:
Technically, this isn't quite true because the select could be edited by someone using dev tools, however, I think this is fine because
1. If you edit a website with dev tools, it's your fault you broke the code, not the code maintainer
2. The likelihood of this happening is next to none

Reply:
Would you be willing to use `OptionType` instead of `T` here? `T` doesn't really mean anything, and while it's a common practice in TypeScript, I see it as equivilent to naming a variable like `const a = 'something'`.

Reply:
I changed them to `V` (for `value` — it's not actually the option type), does that read a bit better? We don't have official naming conventions for this project yet, so eventually we may decide on a prefixed convention like `TValue`. But to write out a full name like `Value`, I think it's harder to signal that it's a generic. And `ValueType` is redundant because everything is a type 😅 We can revisit, but I hope `V` is a good compromise for now.